spark-operator/src/main/java/org/apache/spark/k8s/operator/config/SparkOperatorConf.java

```diff
@@ -0,0 +1,429 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.spark.k8s.operator.config;
+
+import java.time.Duration;
+
+import io.javaoperatorsdk.operator.api.config.LeaderElectionConfiguration;
+import io.javaoperatorsdk.operator.processing.event.rate.LinearRateLimiter;
+import io.javaoperatorsdk.operator.processing.event.rate.RateLimiter;
+import io.javaoperatorsdk.operator.processing.retry.GenericRetry;
+import lombok.extern.slf4j.Slf4j;
+
+import org.apache.spark.k8s.operator.reconciler.SparkReconcilerUtils;
+
+/** Spark Operator Configuration options. */
+@Slf4j
+public class SparkOperatorConf {
+  public static final ConfigOption<String> OPERATOR_APP_NAME =
+      ConfigOption.<String>builder()
+          .key("spark.kubernetes.operator.name")
+          .typeParameterClass(String.class)
+          .description("Name of the operator.")
+          .defaultValue("spark-kubernetes-operator")
+          .enableDynamicOverride(false)
+          .build();
+
+  public static final ConfigOption<String> OPERATOR_NAMESPACE =
+      ConfigOption.<String>builder()
+          .key("spark.kubernetes.operator.namespace")
+          .typeParameterClass(String.class)
+          .description("Namespace that operator is deployed within.")
+          .defaultValue("default")
+          .enableDynamicOverride(false)
+          .build();
+
+  public static final ConfigOption<String> OPERATOR_WATCHED_NAMESPACES =
+      ConfigOption.<String>builder()
+          .key("spark.kubernetes.operator.watchedNamespaces")
+          .description(
+              "Comma-separated list of namespaces that the operator would be "
+                  + "watching for Spark resources. If unset, operator would "
+                  + "watch all namespaces by default.")
+          .defaultValue(null)
```

Comment:
As a continuation of this discussion (https://github.com/apache/spark-kubernetes-operator/pull/12/files#r1641231141), I'd like to recommend the following.
- To have `default` as a `defaultValue` value in the community because I don't think `spark-operator` need to blindly monitor all newly added namespaces.
- Although the comma-separated list is good, we may be able to support a REGEX pattern in general. Or, at least, support a special namespace `*` to monitor all namespaces.

Reply:
To be clear, `.defaultValue("default")` and support "*" is the minimum requirement.

Reply:
While more regex it could become tricky, as k8s API does not allow us to list / watch by patterns yet. Added support for `*` as its more straightforward